compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/body/resolve/FirExpressionsResolveTransformer.kt

```diff
@@ -280,6 +280,31 @@ class FirExpressionsResolveTransformer(transformer: FirBodyResolveTransformer) :
         return resolved.compose()
     }
 
+    override fun transformCheckNotNullCall(
+        checkNotNullCall: FirCheckNotNullCall,
+        data: ResolutionMode
+    ): CompositeTransformResult<FirStatement> {
+        // Resolve the return type of a call to the synthetic function with signature:
+        //   fun <K> checkNotNull(arg: K?): K
+        // ...in order to get the not-nullable type of the argument.
+
+        if (checkNotNullCall.calleeReference is FirResolvedNamedReference && checkNotNullCall.resultType !is FirImplicitTypeRef) {
+            return checkNotNullCall.compose()
+        }
+
+        checkNotNullCall.transformArguments(transformer, ResolutionMode.ContextDependent)
+
+        val result = components.syntheticCallGenerator.generateCalleeForCheckNotNullCall(checkNotNullCall)?.let {
+            callCompleter.completeCall(it, data.expectedType)
+        } ?: run {
+            checkNotNullCall.resultType =
+                FirErrorTypeRefImpl(null, FirSimpleDiagnostic("Can't resolve checkNotNull call", DiagnosticKind.InferenceError))
```

Comment:
Replace error message with `Can't resolve !! operator call`. It prevent possible confuse if there is a regular function named `checkNotNull` in user's project

Reply:
Makes sense. Done.